### PR TITLE
STOR-1596: Bump all deps for 4.17

### PR DESCRIPTION
**Pull request by dfajmon**
Updating all dependencies using go get -u ./....
Update Dockerfile image with golang 1.22 and OCP 4.17

/label px-approved
/label docs-approved

@openshift/storage

### CODE REVIEW

**Comment by dfajmon** [comment]
with the new vendor version of openshift, there is missing a file
`vendor/github.com/openshift/api/sharedresource/v1alpha1/*.crd.yaml`
this makes the tests fail as they can't create a container

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.17 AS builder
 WORKDIR /go/src/github.com/openshift/shared-resources-operator
 COPY . .
-COPY vendor/github.com/openshift/api/sharedresource/v1alpha1/*.crd.yaml assets/
+COPY vendor/github.com/openshift/api/sharedresource/v1alpha1/zz_generated.crd-manifests/*.crd.yaml assets/
```
Comment (jsafrane) [inline Dockerfile:4]: The files have different names now. I would update COPY command here to rename them to the expected names. The operator logs
```
F0805 15:13:16.584002       1 cmd.go:179] error occurred reading file "0000_10_sharedsecret.crd.yaml": open 0000_10_sharedsecret.crd.yaml: file does not exist
```

**Review by jsafrane — COMMENTED** [review]
Please run  `make update-gofmt` to resolve verification errors.

**Comment by dobsonj** [comment]
/assign @gabemontero 
for `/approve`

**Comment by gabemontero** [comment]
/assign @sayan-biswas 
/assign @adambkaplan 

/unassign @gabemontero 

Sayan and Adam have taken over csi @dobsonj  - thanks

**Comment by jsafrane** [comment]
/label qe-approved
CI is green

**Review by sayan-biswas — APPROVED** [review]

**Pull request merged**